src/platform/qpg6100/qpg6100Config.cpp

```diff
@@ -46,106 +46,78 @@ CHIP_ERROR QPG6100Config::Init()
 
 uint16_t QPG6100Config::GetSettingsMaxValueLength(Key key)
 {
-    return qvCHIP_Nvm_GetMaxKeyLen(key);
+    uint16_t keyLen;
+
+    return (qvCHIP_Nvm_GetMaxKeyLen(key, &keyLen) == QV_STATUS_NO_ERROR) ? keyLen : 0;
 }
 
 CHIP_ERROR QPG6100Config::ReadConfigValue(Key key, bool & val)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
     uint16_t length;
-    bool res;
 
     length = sizeof(bool);
 
-    res = qvCHIP_Nvm_Restore(key, (uint8_t *) (&val), &length);
-    if (res == false)
-    {
-        err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
-    }
-
-    return err;
+    return MapNVMError(qvCHIP_Nvm_Restore(key, reinterpret_cast<uint8_t *>(&val), &length));
 }
 
 CHIP_ERROR QPG6100Config::ReadConfigValue(Key key, uint32_t & val)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
     uint16_t length;
-    bool res;
 
     length = sizeof(uint32_t);
 
-    res = qvCHIP_Nvm_Restore(key, (uint8_t *) (&val), &length);
-    if (res == false)
-    {
-        err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
-    }
-
-    return err;
+    return MapNVMError(qvCHIP_Nvm_Restore(key, reinterpret_cast<uint8_t *>(&val), &length));
 }
 
 CHIP_ERROR QPG6100Config::ReadConfigValue(Key key, uint64_t & val)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
     uint16_t length;
-    bool res;
 
     length = sizeof(uint64_t);
 
-    res = qvCHIP_Nvm_Restore(key, (uint8_t *) (&val), &length);
-    if (res == false)
-    {
-        err = CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
-    }
-
-    return err;
+    return MapNVMError(qvCHIP_Nvm_Restore(key, reinterpret_cast<uint8_t *>(&val), &length));
 }
 
 CHIP_ERROR QPG6100Config::ReadConfigValueStr(Key key, char * buf, size_t bufSize, size_t & outLen)
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
-    bool res;
+    qvStatus_t res;
     uint16_t length;
 
     if (buf == NULL)
     {
         return CHIP_ERROR_INVALID_ARGUMENT;
     }
 
-    length = (uint16_t) bufSize;
-    res    = qvCHIP_Nvm_Restore(key, (uint8_t *) buf, &length);
+    length = static_cast<uint16_t>(bufSize);
```

Comment:
What happens if this overflows?  Should we not have a range check before casting?